lucene/core/src/test/org/apache/lucene/search/TestConjunctionDISI.java

```diff
@@ -391,4 +391,20 @@ public void testCollapseSubConjunctionDISIs() throws IOException {
   public void testCollapseSubConjunctionScorers() throws IOException {
     testCollapseSubConjunctions(true);
   }
+
+  public void testIllegalAdvancementOfSubIteratorsTripsAssertion() throws IOException {
+    int maxDoc = 100;
```

Comment:
I think we need to call something like `assumeTrue(assertsAreEnabled)` since this test requires assertions to be enabled to succeed.

Reply:
Thanks @jpountz, very good point, addressed in bf9a3e8